homeassistant/components/tractive/__init__.py

```diff
@@ -145,13 +150,11 @@ async def _generate_trackables(
     return Trackables(tracker, trackable, tracker_details, hw_info, pos_report)
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: TractiveConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
-        tractive = hass.data[DOMAIN][entry.entry_id].pop(CLIENT)
-        await tractive.unsubscribe()
-        hass.data[DOMAIN].pop(entry.entry_id)
+        await entry.runtime_data.client.unsubscribe()
```

Comment:
You could you `entry.async_on_unload(tractive.unsubscribe)` in `async_setup_entry` for that.